lib/src/main/java/org/team1540/rooster/drive/pipeline/SimpleJoystickInput.java

```diff
@@ -0,0 +1,132 @@
+package org.team1540.rooster.drive.pipeline;
+
+import edu.wpi.first.wpilibj.Joystick;
+import java.util.OptionalDouble;
+import org.team1540.rooster.Utilities;
+
+/**
+ * Simple tank-style input from a WPILib {@link Joystick}. The left and right joysticks are used to
+ * control the left and right sides of the robot respectively. Additionally, there is an optional
+ * "forward axis" and "back axis"; if the forward axis is bound to one trigger on an XBox controller
```

Comment:
```suggestion
 * "forward axis" and "back axis"; if the forward axis is bound to one trigger on an Xbox controller
```

Reply:
That said, it's not really specific to an Xbox controller, is it? Also ensure that there's checking that the joystick doesn't go negative all of a sudden. Looks like there is, but I'm not 100% sure about my reading of your constraints.

Reply:
What do you mean checking the joystick doesn't go negative?